backend/app/api/api_v1/endpoints/projects.py

```diff
@@ -87,6 +107,64 @@ async def update_json_data_for_prefix_in_project(
     return JSONDataOut(can_edit=True, data=data, task_id=task_id)
 
 
+@router.post(
+    "/create/",
+    status_code=status.HTTP_201_CREATED,
+    dependencies=[Depends(is_admin_or_superuser), Depends(is_user_active)],
+)
+async def create_new_project(
+    current_user: Annotated[UserBase, Depends(utils.get_current_user)],
+    user_github_id: int,
+    root_path: Path,
```

Comment:
1. Currently the path needs to start with `checkouts/unpublished/` to be considered valid data. The frontend shouldn't care about this, nor should it be aware of the internal structure of the backend. The endpoint should only accept paths that start with `root/`.
2. When I provide a path to an `﻿html` or any other type of file, the endpoint throws a ﻿`500` status code with an `﻿Internal Server Error`. Please add endpoint-wide validation for the ﻿root_path (e.g., in the form of dependency injection) that checks whether the provided path is valid for this endpoint. This validation should ensure that the path exists and is a root path. If the path is invalid, provide the user with information about what went wrong instead of raising a ﻿`ValueError` in the utils.